CONTRIBUTING.md

```diff
@@ -1,59 +1,604 @@
+and the E2E tests you can get to work :tada: .
+
+1. First create a topic branch from where you want to base your work (this is usually
+  from `main`):
+
+      ```bash
+      git checkout -b <feature-name>
+      ```
+
+1. Write your solution. Try to align with existing patterns and standards.
+
+1. Try to commit in small chunks so that changes are well described
+  and not lumped in with something unrelated. This will make debugging easier in
+  the future.
+  Make sure commit messages are in the [proper format](#commit-message-formatting).
+
+1. Ensure that every commit is **compilable** and that the tests run successfully at
+  that point.
+
+1. Make sure your commits are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits).
+
+1. **All code contributions must be submitted with unit tests**. See [this package](https://github.com/weaveworks/flintlock/tree/main/core/steps/microvm)
+  package for a good example of tests.
+
+1. For extra special bonus points, if you see any tests missing from the area you are
+  working on, please add them! It will be much appreciated :heart: .
+
+1. Check the documentation and update it to cover your changes,
+  either in the [README](README.md), [userdocs](userdocs) or [docs](docs/) folder.
+  If you have added a completely new feature please ensure that it is documented
+  thoroughly.
+
+1. Before you [open your PR](#pr-submission-guidelines), run all the unit and E2E tests and manually
+  verify that your solution works.
+
+## Asking for help
+
+If you need help at any stage of your work, please don't hesitate to ask!
+
+- To get more detail on the issue you have chosen, it is a good idea to start by asking
+  whoever created it to provide more information.
+  If they do not respond, or more help is needed,
+  you can then bring in one of the [core maintainers](MAINTAINERS).
+
+- If you are struggling with something while working on your PR, or aren't quite
+  sure of your approach, you can open a [draft](https://github.blog/2019-02-14-introducing-draft-pull-requests/)
+  (prefix the title with `WIP: `) and explain what you are thinking.
+  You can also tag in one of the core team, or drop the PR link into [slack][slack-channel].
+
+- We are happy to pair with contributors over a slack call to help them fine-tune their
+  implementation. You can ping us directly, or head to the [channel][slack-channel]
+  to see if anyone in the community is up for being a buddy :smiley: .
+
+## PR submission guidelines
+
+Push your changes to the branch on your fork and submit a pull request to the original repository
+against the `main` branch.
+Where possible, please squash your commits to ensure a tidy and descriptive history.
+
+```bash
+git push <remote-name> <feature-name>
+```
+
+If your PR is still a work in progress, please open a [Draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/).
```

Comment:
Is it worth mentioning the `WIP` prefix again?

Reply:
definitely 👍